*** numba/setup.py-dist	2021-12-14 15:41:49.122437547 +0900
--- numba/setup.py	2021-12-14 15:41:57.569244516 +0900
***************
*** 20,26 ****
  
  
  min_python_version = "3.7"
! max_python_version = "3.10"  # exclusive
  min_numpy_build_version = "1.11"
  min_numpy_run_version = "1.17"
  max_numpy_run_version = "1.21"
--- 20,26 ----
  
  
  min_python_version = "3.7"
! max_python_version = "3.11"  # exclusive
  min_numpy_build_version = "1.11"
  min_numpy_run_version = "1.17"
  max_numpy_run_version = "1.21"
*** numba/numba/_dispatcher.cpp-dist	2021-12-14 15:51:53.046336957 +0900
--- numba/numba/_dispatcher.cpp	2021-12-14 15:55:29.450842421 +0900
***************
*** 25,33 ****
      if (tstate->tracing)
          return 0;
      tstate->tracing++;
!     tstate->use_tracing = 0;
      result = func(obj, frame, what, arg);
!     tstate->use_tracing = ((tstate->c_tracefunc != NULL)
                             || (tstate->c_profilefunc != NULL));
      tstate->tracing--;
      return result;
--- 25,33 ----
      if (tstate->tracing)
          return 0;
      tstate->tracing++;
!     tstate->cframe->use_tracing = 0;
      result = func(obj, frame, what, arg);
!     tstate->cframe->use_tracing = ((tstate->c_tracefunc != NULL)
                             || (tstate->c_profilefunc != NULL));
      tstate->tracing--;
      return result;
***************
*** 384,390 ****
      fn = (PyCFunctionWithKeywords) PyCFunction_GET_FUNCTION(cfunc);
      tstate = PyThreadState_GET();
  
!     if (tstate->use_tracing && tstate->c_profilefunc)
      {
          /*
           * The following code requires some explaining:
--- 384,390 ----
      fn = (PyCFunctionWithKeywords) PyCFunction_GET_FUNCTION(cfunc);
      tstate = PyThreadState_GET();
  
!     if (tstate->cframe->use_tracing && tstate->c_profilefunc)
      {
          /*
           * The following code requires some explaining:
***************
*** 660,666 ****
       * not compile one */
      int exact_match_required = self->can_compile ? 1 : self->exact_match_required;
  
!     if (ts->use_tracing && ts->c_profilefunc) {
          locals = PyEval_GetLocals();
          if (locals == NULL) {
              goto CLEANUP;
--- 660,666 ----
       * not compile one */
      int exact_match_required = self->can_compile ? 1 : self->exact_match_required;
  
!     if (ts->cframe->use_tracing && ts->c_profilefunc) {
          locals = PyEval_GetLocals();
          if (locals == NULL) {
              goto CLEANUP;
***************
*** 774,780 ****
       * not compile one */
      int exact_match_required = self->can_compile ? 1 : self->exact_match_required;
  
!     if (ts->use_tracing && ts->c_profilefunc) {
          locals = PyEval_GetLocals();
          if (locals == NULL) {
              goto CLEANUP;
--- 774,780 ----
       * not compile one */
      int exact_match_required = self->can_compile ? 1 : self->exact_match_required;
  
!     if (ts->cframe->use_tracing && ts->c_profilefunc) {
          locals = PyEval_GetLocals();
          if (locals == NULL) {
              goto CLEANUP;
